runtime/src/iree/vm/ref_test.cc

```diff
@@ -166,6 +166,32 @@ TEST(VMRefTest, WrappingReleasesExisting) {
   iree_vm_ref_release(&ref);
 }
 
+// Tests that wrapping releases any existing ref in out_ref.
+TEST(VMRefTest, WrappingRetainExisting) {
```

Comment:
ASan failure in here: https://github.com/iree-org/iree/actions/runs/9421503282/job/25955732007?pr=17610#step:4:11982

```
 ==73699==ERROR: LeakSanitizer: detected memory leaks
  
  Direct leak of 8 byte(s) in 1 object(s) allocated from:
      #0 0x5636df6b2cfd in operator new(unsigned long) (/work/build-asan/runtime/src/iree/vm/ref_test+0x128cfd) (BuildId: 3b81e242324e14a2a00840a817ed79f6f1280fb7)
      #1 0x5636df6b781a in (anonymous namespace)::VMRefTest_WrappingRetainExisting_Test::TestBody() /work/runtime/src/iree/vm/ref_test.cc:174:27
      #2 0x5636df768ea0 in void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) /work/third_party/googletest/googletest/src/gtest.cc:2635:10
      #3 0x5636df768ea0 in void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) /work/third_party/googletest/googletest/src/gtest.cc:2671:14
      #4 0x5636df721d83 in testing::Test::Run() /work/third_party/googletest/googletest/src/gtest.cc:2710:5
      #5 0x5636df726644 in testing::TestSuite::Run() /work/third_party/googletest/googletest/src/gtest.cc:3034:30
      #6 0x5636df7507a7 in testing::internal::UnitTestImpl::RunAllTests() /work/third_party/googletest/googletest/src/gtest.cc:5964:44
      #7 0x5636df76b310 in bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) /work/third_party/googletest/googletest/src/gtest.cc:2635:10
      #8 0x5636df76b310 in bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) /work/third_party/googletest/googletest/src/gtest.cc:2671:14
      #9 0x5636df74fa62 in testing::UnitTest::Run() /work/third_party/googletest/googletest/src/gtest.cc:5543:10
      #10 0x5636df6db5d3 in RUN_ALL_TESTS() /work/third_party/googletest/googletest/include/gtest/gtest.h:2334:73
      #11 0x5636df6db5d3 in main /work/runtime/src/iree/testing/gtest_main.cc:20:13
      #12 0x7ff0138bfd8f  (/lib/x86_64-linux-gnu/libc.so.6+0x29d8f) (BuildId: c289da5071a3399de893d2af81d6a30c62646e1e)
  
  SUMMARY: AddressSanitizer: 8 byte(s) leaked in 1 allocation(s).
```

Reply:
hooray asan!